phrwatcher.php

```diff
@@ -22,6 +22,12 @@
 	$ENABLED = true;
 
 
+	/**
+	 * For additional security, input your development site address, to recognize in case of accidental migration into production
+	 */
+	if ( ! in_array( $_SERVER['HTTP_HOST'], ['localhost', '127.0.0.1', '::1', 'test.example.com'] ) ) 
```

Comment:
What if we create a $DISABLED_HOSTS after the $DISABLED variable, then add your snippet to the "dont edit area"?

 ```php
 	$DISABLED_HOSTS = [
 		"your-production-domain.com"
 	];
 	
 	// ---------------------- Dont Edit It ----------------------
 
 	if (in_array($_SERVER['HTTP_HOST'],  $DISABLED_HOSTS)) {
 	 	exit("The PHP Hot Reloader was disabled for \"{$_SERVER['HTTP_HOST']}\". This host is in $DISABLED_HOSTS list");
 	}
 ```

Reply:
Yes, sure, my example was just concept, an d actually meant what you say (to be stored in varaible).
However, I think that I might have not expressed it correctly and you misunderstood. The "disabled" host is not desired solution. In opposite, we need to say "Enabled Hosts", so the developer (which works surely on specific development stage) will say that "I enable this script to work only on LOCALHOST[or whatever] ", i.e:

	$ENABLED_HOSTS = [
		'localhost', '127.0.0.1', 
		'::1', 
		'test.example.com'
	];
	
	// ---------------------- Dont Edit It ----------------------

	if ( !in_array($_SERVER['HTTP_HOST'],  $ENABLED_HOSTS )) {
	 	exit("The PHP Hot Reloader was disabled for \"{$_SERVER['HTTP_HOST']}\". This host is not in $ENABLED_HOSTS list");
	}

Disabled_Hosts, on other hand of misuse, doesn't carry any meaning in most cases, when the production link is not even know in advance, and developer just work on script which will be deployed on "noone knows in advance" production domain.  
So, at least these 2 arguments could be enough to avoid DISABLED approach, and in opposite, implement "ENABLED" ones.

Reply:
Actually, even though either $ENABLED_HOSTS or $DISABLED_HOSTS are quite good layer of security (in terms of the meant accidental upload or exposure), they are not still full protection, as they are vulnerable by IP-mapping, if site is hosted on dedicated IP (VPS, DEDICATED server), where IP address is being solved to only 1 specific site. 
In that case, both of them can be bypassed. 
However, in many cases (might be 90%+ imho) this is not case, and thus, this proposed update is quite good improvement in protection.

Reply:
I've extended the concept I wanted to PR. 

In addition to addressing this exact PR, I've in addition, moved the "DO NOT EDIT" parts into the base-file itself.
I absolutely keen that the base functionality (which customer is not meant to edit when using script out-of-the-box) should not be in adoptable file. All those fixed functionalities (which are not meant for editing to users) should be moved into the base file, so, as you will check, that approach might have more sense.  It's better to do those checks and validations there, and in `phwachter` just include that base file. that's all simple.
Let me know.

Reply:
hey, really nice! im seriously busy this week :/ so ill take a look on the PR this weekend, im sending this to justify some gaps to answear, so sorry in advance, and thanks for your PR.